numba/mviewbuf.c

```diff
@@ -1,8 +1,22 @@
 #include "_pymodule.h"
 
-static int get_buffer(PyObject* obj, Py_buffer *buf)
+static int get_buffer(PyObject* obj, Py_buffer *buf, int force)
 {
-    return PyObject_GetBuffer(obj, buf, PyBUF_WRITABLE|PyBUF_ND|PyBUF_STRIDES|PyBUF_FORMAT);
+    Py_buffer read_buf;
+    int flags = PyBUF_ND|PyBUF_STRIDES|PyBUF_FORMAT;
+
+    /* Attempt to get a writable buffer */
+    if (!PyObject_GetBuffer(obj, buf, flags|PyBUF_WRITABLE))
+        return 0;
+    if (!force)
+        return -1;
+
+    /* Make a writable buffer from a read-only buffer */
+    PyErr_Clear();
+    if(-1 == PyObject_GetBuffer(obj, &read_buf, flags))
+        return -1;
+    return PyBuffer_FillInfo(buf, NULL, read_buf.buf, read_buf.len, 0,
```

Comment:
Any reason you don't simply flip the "readable" switch in the buffer?
Also note that read_buf must be released using PyBuffer_Release().